src/Glpi/Config/LegacyConfigurators/BaseConfig.php

```diff
@@ -0,0 +1,274 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Config\LegacyConfigurators;
+
+use Glpi\Config\LegacyConfigProviderInterface;
+use Symfony\Component\DependencyInjection\Attribute\Autowire;
+use Symfony\Component\HttpFoundation\Request;
+
+final readonly class BaseConfig implements LegacyConfigProviderInterface
+{
+    public function __construct(
+        #[Autowire('%kernel.project_dir%')] private string $projectDir,
+    ) {
+    }
+
+    public function execute(Request $request): void
+    {
+        if (!defined('GLPI_ROOT')) {
+            define('GLPI_ROOT', $this->projectDir);
+        }
```

Comment:
It should not be defined anywhere else. If it does, it must fail so we can detect other unexpected declarations.
```suggestion
        define('GLPI_ROOT', $this->projectDir);
```